test/metabase/search/filter_test.clj

```diff
@@ -76,26 +78,63 @@
             [:= :card.archived false]]
            (:where (search.filter/build-filters
                     base-search-query "card"
-                    (merge default-search-ctx {:search-string "a string"}))))))
-
+                    (merge default-search-ctx {:search-string "a string"})))))))
+
+(deftest  build-created-at-filter-test
+  (testing "created-at filter"
+    (mt/with-clock #t "2023-05-04T10:00:05Z[UTC]"
+      (are [created-at expected-where]
+           (= expected-where
+              (->> (search.filter/build-filters
+                    base-search-query "card"
+                    (merge default-search-ctx {:created-at created-at}))
+               :where
+               ;; drop the first 2 clauses [:and [:= :card.archived false]]
+               (drop 2)))
+           ;; absolute datetime
+           "Q1-2023"                                 [[:>= [:cast :card.created_at :date] #t "2023-01-01"]
+                                                      [:< [:cast :card.created_at :date]  #t "2023-04-01"]]
+           "2016-04-18~2016-04-23"                   [[:>= [:cast :card.created_at :date] #t "2016-04-18"]
+                                                      [:< [:cast :card.created_at :date]  #t "2016-04-24"]]
+           "2016-04-18T10:00:00~2016-04-23T11:00:00" [[:>= :card.created_at #t "2016-04-18T00:00"]
+                                                      [:< :card.created_at #t "2016-04-24T00:00"]]
+
+           "~2023-05-04"                             [[:< [:cast :card.created_at :date]  #t "2023-05-05"]]
+           "2023-05-04~"                             [[:> [:cast :card.created_at :date]  #t "2023-05-04"]]
+           ;; relative datetime
+           "past3days"                               [[:>= [:cast :card.created_at :date] #t "2023-05-01"]
```

Comment:
Can we also add a test for a range with an inclusive end? e.g. "past3days~"
And another test for inclusive end with a unit other than days, e.g. "past3hours~"